python/taichi/misc/gui.py

```diff
@@ -47,32 +49,71 @@ def clear(self, color=None):
             color = self.background_color
         self.canvas.clear(color)
 
-    def set_image(self, img):
-        import numpy as np
-        from .image import cook_image
-        img = cook_image(img)
+    def cook_image(self, img):
         if img.dtype in [np.uint8, np.uint16, np.uint32, np.uint64]:
             img = img.astype(np.float32) * (1 / np.iinfo(img.dtype).max)
         elif img.dtype in [np.float32, np.float64]:
-            img = np.clip(img.astype(np.float32), 0, 1)
+            img = img.astype(np.float32)
+            from .util import get_os_name
+            if img.dtype == np.float32 and get_os_name() == 'win':
```

Comment:
My previous comment still stands here: if we can't assume the operating system will do the clamping in all circumstances, we'd better keep the consistency and clamp it ourselves no matter what.

Reply:
Thank for the smart idea! but I just figured out how to make win clamp too, no worry now!